From 2bdaa7d3bfa5193554a7d4d24a51dde376f3d70c Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:40:13 -0400
Subject: [ethernet] i40e: Fix firmware API version errors

Message-id: <1409658042-22689-253-git-send-email-sassmann@redhat.com>
Patchwork-id: 90435
O-Subject: [RHEL7.1 PATCH 252/281] i40e: Fix firmware API version errors
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Catherine Sullivan <catherine.sullivan@intel.com>

Reword the error messages. Also add a major version check because
We only want to warn on nvm_minor > expected_minor if
nvm_major == expected_major. Lastly, change an if to an else if
because the two statements will never evaluate to true at the same time.

Change-ID: I6ddf9986f26b35f6879cbeac4fcef04a8497a383
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Jim Young <jamesx.m.young@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 7aa67613172734d5d6be99db019e57e453f76862)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index dc0d32b..ad56128 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -8642,24 +8642,18 @@ static int i40e_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
  dev_info(&pdev->dev, "%s\n", i40e_fw_version_str(hw));
  if (err) {
   dev_info(&pdev->dev,
-    "init_adminq failed: %d expecting API %02x.%02x\n",
-    err,
-    I40E_FW_API_VERSION_MAJOR, I40E_FW_API_VERSION_MINOR);
+    "The driver for the device stopped because the NVM image is newer than expected. You must install the most recent version of the network driver.\n");
   goto err_pf_reset;
  }
 
- if (hw->aq.api_min_ver > I40E_FW_API_VERSION_MINOR)
+ if (hw->aq.api_maj_ver == I40E_FW_API_VERSION_MAJOR &&
+     hw->aq.api_min_ver > I40E_FW_API_VERSION_MINOR)
   dev_info(&pdev->dev,
-    "Note: FW API version %02x.%02x newer than expected %02x.%02x, recommend driver update.\n",
-    hw->aq.api_maj_ver, hw->aq.api_min_ver,
-    I40E_FW_API_VERSION_MAJOR, I40E_FW_API_VERSION_MINOR);
-
- if (hw->aq.api_maj_ver < I40E_FW_API_VERSION_MAJOR ||
-     hw->aq.api_min_ver < (I40E_FW_API_VERSION_MINOR-1))
+    "The driver for the device detected a newer version of the NVM image than expected. Please install the most recent version of the network driver.\n");
+ else if (hw->aq.api_maj_ver < I40E_FW_API_VERSION_MAJOR ||
+   hw->aq.api_min_ver < (I40E_FW_API_VERSION_MINOR - 1))
   dev_info(&pdev->dev,
-    "Note: FW API version %02x.%02x older than expected %02x.%02x, recommend nvm update.\n",
-    hw->aq.api_maj_ver, hw->aq.api_min_ver,
-    I40E_FW_API_VERSION_MAJOR, I40E_FW_API_VERSION_MINOR);
+    "The driver for the device detected an older version of the NVM image than expected. Please update the NVM image.\n");
 
 
  i40e_verify_eeprom(pf);
-- 
1.7.1